ext/liquid_c/vm.c

```diff
@@ -336,7 +341,13 @@ static VALUE vm_render_until_error(VALUE uncast_args)
                     num_args = *ip++; // includes input argument
                 }
 
-                VALUE *args_ptr = vm_stack_pop_n_use_in_place(vm, num_args);
+                VALUE *popped_args_ptr = vm_stack_pop_n(vm, num_args);
+                /* We have to copy popped_args_ptr to the stack because the VM
+                 * no longer holds onto these objects, so they have to exist on
+                 * the stack to ensure they don't get garbage collected. */
+                VALUE *args_ptr = alloca(sizeof(VALUE *) * num_args);
+                memcpy(args_ptr, popped_args_ptr, sizeof(VALUE *) * num_args);
+
                 VALUE result = vm_invoke_filter(vm, filter_name, num_args, args_ptr);
```

Comment:
Why can't this also use `vm_stack_peek_n` and `vm_stack_pop_n`?

```suggestion
                VALUE *args_ptr = vm_stack_peek_n(vm, num_args);
                VALUE result = vm_invoke_filter(vm, filter_name, num_args, args_ptr);
                vm_stack_pop_n(vm, num_args);
```

Reply:
I was worried about the filter called from `vm_invoke_filter` appending onto the stack of the current VM and causing issues, but now that I think about it again, that should be fine right?